python/tk_nuke/menu_generation.py

```diff
@@ -21,291 +18,665 @@
 import nukescripts.openurl
 import nukescripts
 
+from PySide import QtGui
+
+# -----------------------------------------------------------------------------
 
 class MenuGenerator(object):
     """
-    Menu generation functionality for Nuke
+    The factory base class for Nuke based menu generators. Since Nuke
+    now provides multiple modes (Nuke, Nuke Studio, and Hiero), we are
+    required to supply menu generation logic for each. Instantiating
+    this class directly will provide the correct implementation for the
+    currently running Nuke mode.
     """
+    def __new__(cls, *args, **kwargs):
```

Comment:
we are doing `if self.hiero` in many places in the engine already. Why not just have this as an (yet another) if statement in the engine instead? That seems clearer to me.